File: packages/plugin/overlay-api/src/main/java/com/overlay/api/communication/types/IncomingPacket.java
package com.overlay.api.communication.types;

import com.eu.habbo.habbohotel.gameclients.GameClient;

public abstract class IncomingPacket<T extends IncomingMessage> {
  public final Class<T> type;

  protected IncomingPacket(Class<T> type) {
    this.type = type;
  }

  public abstract void handle(GameClient client, T message);
}

File: packages/plugin/overlay-plugin/src/main/java/io/github/brenoepics/overlay/plugin/messages/incoming/IncomingPacket.kt
package io.github.brenoepics.overlay.plugin.messages.incoming

import com.eu.habbo.habbohotel.gameclients.GameClient
import io.github.brenoepics.overlay.api.communication.types.ApiIncomingPacket
import io.github.brenoepics.overlay.api.communication.types.IncomingMessage

abstract class IncomingPacket<T : IncomingMessage>(type: Class<T>) : ApiIncomingPacket<T, GameClient>(type)